lib/nats/io/client.rb

```diff
@@ -167,7 +170,7 @@ def initialize(uri = nil, opts = {})
       @uri = nil
       @server_pool = []
 
-      @status = DISCONNECTED
+      @status = NOT_YET_CONNECTED
```

Comment:
I think here we can declare as `@status = nil` and then at the beginning of `connect` then set it as `DISCONNECTED`